protocols/shell-workspace-creation-v1.xml

```diff
@@ -0,0 +1,321 @@
+        Destroys the object.
+      </description>
+    </request>
+  </interface>
+
+  <interface name="shell_workspace_target_creation_manager_v1" version="1">
+    <request name="destroy" type="destructor">
+      <description summary="destroy the manager">
+        Destroy the manager. Destroying the manager does not destroy objects
+        created by the manager.
+      </description>
+    </request>
+
+    <request name="get_target">
+      <description>
+        This request creates a new workspace target from a given workspace
+        target descriptor.
+
+        The descriptor may be destroyed following this request.
+
+        Because the target descriptor cannot be versioned, this request
+        is used to create a versionable target from the descriptor.
+      </description>
+
+      <arg name="target" type="new_id" interface="shell_workspace_creation_target_v1"/>
+      <arg name="descriptor" type="new_id" interface="shell_workspace_creation_target_descriptor_v1"/>
+    </request>
+  </interface>
+
+  <interface name="shell_workspace_arbitrary_creation_manager_v1" version="1">
+    <request name="destroy" type="destructor">
+      <description summary="destroy the manager">
+        Destroy the manager. Destroying the manager does not destroy objects
+        created by the manager.
+      </description>
+    </request>
+
+    <request name="get_arbitrary_target">
+      <description summary="create a workspace target relative to another workspace">
+        This requests creates a workspace target that targets an arbitrary
+        compositor chosen name and location for the workspace.
+      </description>
+
+      <arg name="descriptor" type="new_id" interface="shell_workspace_creation_target_descriptor_v1"/>
+    </request>
+  </interface>
+
+  <interface name="shell_workspace_coordinate_creation_manager_v1" version="1">
+    <request name="destroy" type="destructor">
+      <description summary="destroy the manager">
+        Destroy the manager. Destroying the manager does not destroy objects
+        created by the manager.
+      </description>
+    </request>
+
+    <request name="get_coordinate_target">
+      <description summary="create a workspace target at the given coordinates">
+        This request creates a workspace target at the given coordinates in
+        the compositor's workspace coordinate space. This coordinate space
+        should match the one used in 'ext_workspace_handle_v1.coordinates'.
+
+        A given set of coordinates may be unrepresentable or otherwise disallowed
+        by the compositor, in which case the workspace target will be invalid.
+
+        If a workspace already exists at the given coordinates, the resuling
+        target should be valid and refer to the existing workspace.
+      </description>
+
+      <arg name="descriptor" type="new_id" interface="shell_workspace_creation_target_descriptor_v1"/>
+      <arg name="coordinates" type="array"/>
```

Comment:
How does this work regarding different outputs for compositors with per-output workspaces? Also: maybe worth adding optional output to get_arbitrary_target?

Reply:
Good spot. It would make sense to use ext-workspace's workspace groups for this. To me it reads like the compositor should either send one per monitor or one for all monitors depending what it supports.